scvi/model/base/_vaemixin.py

```diff
@@ -177,3 +177,7 @@ def get_latent_representation(
 
             latent += [z.cpu()]
         return torch.cat(latent).numpy()
+
+    @torch.no_grad()
+    def get_batch_embedding(self):
```

Comment:
Need typing on the return. It seems like it would be a torch object which is not preferable at the model abstraction level. Convert to a numpy array before returning.

Reply:
Oh now i see whats going on with this torch.nn.Embedding typing. I think it would be more natural to take batch indices as a parameter and return the relevant embeddings to the user. If the indices=None then you just return the embedding for each of the batches used to train the model. Generally better to avoid returning torch level objects at the model abstraction level (allows for different backends, keeps user level backend agnostic, avoids issues with memory device management)